### DISCUSSION TITLE
First-boot configuration

### DISCUSSION THREAD

**Discussion by daenney** [gokrazy/gokrazy#201] [Q&A] [answered]
I've been looking at using gokrazy to provide an image anyone can flash on a Pi to make it easy to deploy a particular application (basically an appliance).

I can get 99% of the way there this way since I don't have to worry about WiFi support in this case. The application I want to run on it I can fully preconfigure except for 1 thing: I need to provide it with a DNS name under which the application will be externally reachable. This value can be set in the config file, but can also be provided as an environment variable. That DNS name will be different from person to person and thus something I'd need them to somehow provide, but ideally without them having to learn to use the gokrazy tools and customise the image that way.

I'm wondering if anyone's run into this sort of "one time configuration after first boot" issue and what solutions might've been used? Ultimately I can always store the configuration on `/perm` and tell folks to edit it that way, but I'm wondering if there's better ways?

**Comment by stapelberg** [comment, marked as answer]
The planned imaged server can be used for precisely such customization, either on behalf of a user (server-side), or with a form in which the user can customize things.

I just made the design doc public: https://docs.google.com/document/d/1a7TXSxNgdR9pkz2KdZxG_gF01DTkneNwtDxLDXh8tkM/edit?usp=sharing

There’s also a Git repository at https://github.com/gokrazy/imaged

It’s not finished yet (and I’m pretty busy currently, so unlikely to see much work from my side), but maybe you can take this and run with it?

**Reply by daenney** [reply]
Well, excellent timing it seems :smile:. Right now the software I'm looking at wrapping up is very much alpha still, so the burden of "edit config directly on SD-card" is pretty low as we expect users to be technical enough to pull that off.

I'll definitely take a look at imaged, sounds like it does pretty much exactly what's needed.
